l10n_ch_import_cresus/wizard/l10n_ch_import_cresus.py

```diff
@@ -137,194 +71,150 @@ def _parse_csv(self):
+                    journal_id=journal_id)
+                lines = []
+            previous_pce = line_cresus['pce']
+            previous_date = line_cresus['date']
+
+            recto_amount = float(line_cresus['amount'].replace('\'', '')
+                                                      .replace(' ', ''))
+            verso_amount = 0.0
+            if recto_amount < 0:
+                recto_amount, verso_amount = 0.0, -recto_amount
+            if line_cresus['debit'] != '...':
+                account = self._find_account(line_cresus['debit'])
+                tax = self._find_tax(line_cresus['typtvat'], account)
+                analytic_account = self._find_analytic_account(
+                    line_cresus['analytic_account'], account)
+                lines.append(self.make_line(
+                    recto_amount, verso_amount,
+                    account_id=account.id,
+                    partner_id=False,
+                    name=line_cresus['ref'],
+                    tax_line_id=tax.id,
+                    analytic_account_id=analytic_account.id))
+
+            if line_cresus['credit'] != '...':
+                account = self._find_account(line_cresus['credit'])
+                tax = self._find_tax(line_cresus['typtvat'], account)
+                analytic_account = self._find_analytic_account(
+                    line_cresus['analytic_account'], account)
+                lines.append(self.make_line(
+                    verso_amount, recto_amount,
+                    account_id=account.id,
+                    partner_id=False,
+                    name=line_cresus['ref'],
+                    tax_line_id=tax.id,
+                    analytic_account_id=analytic_account.id))
+
+        yield self.make_move(
+            lines,
+            date=line_cresus['date'],
+            ref=previous_pce,
+            journal_id=journal_id)
 
-            self._manage_load_results(res)
-        except Exception as exc:
-            ex_type, sys_exc, tb = sys.exc_info()
-            tb_msg = ''.join(traceback.format_tb(tb, 30))
-            _logger.error(tb_msg)
-            _logger.error(repr(exc))
-            self.report = _("Unexpected exception.\n %s \n %s" %
-                            (repr(exc), tb_msg))
-            self.state = 'error'
-        finally:
-            if self.state == 'error':
-                self.env.cr.rollback()
-                self.write({'report': self.report, 'state': self.state})
-        return {}
+    @api.multi
+    def _import_file(self):
+        move_obj = self.env['account.move']
+        data = self._parse_csv()
+        data = self._standardise_data(data)
+        self.imported_move_ids = [move_obj.create(mv).id for mv in data]
 
     @api.multi
     def import_file(self):
-        data = self._parse_csv()
-        new_data = self._standardise_data(data)
-        return self._load_data(new_data)
+        try:
+            self._import_file()
+        except Exception as exc:
```

Comment:
The point is to catch any unexpected error and show it in our interface rather that the default "Odoo Server Error" popup. Got it this way from 8.0.